graphql-dgs/src/main/kotlin/com/netflix/graphql/dgs/internal/DgsSchemaProvider.kt

```diff
@@ -241,7 +241,7 @@ class DgsSchemaProvider(private val applicationContext: ApplicationContext,
                             } catch (ex: Exception) {
                                 throw DgsInvalidInputArgumentException("Specified type '${collectionType}' is invalid for $parameterName.", ex)
                             }
-                        } else if (parameterValue is MultipartFile) {
+                        } else if (parameterValue is List<*> || parameterValue is MultipartFile) {
```

Comment:
Shouldn't this be part of the next if statement, and check the type of the the type argument?
The Multipartfile check is a very specific case to check for, so I think combining that with other use cases will be hard to understand later.

Reply:
Added a separate case for it because it does handle other List types as well..list of custom scalars, list of Integers etc.